### PR TITLE
Ticket 1620/resolve speed issues of ima event scanner

### LINKED ISSUE #1620: IMA: getPastEvents sometimes work significantly slow as expected
**Description** 
Case: compatibility check 
IMA version 11/16: 2.0.0
IMA version 5/16: 1.5.0-beta.1
Skaled: the same for all - 3.16.1

**Step to reproduce**
Run transfers loop between schain
When processing message takes more than 60 seconds observe logs on working node

**Actual result** 
In some regular use case search of IMA msg can takes more then 30-... min, depends on amount of blocks on the schain. 
NOTE: Root cause investigate in progress 

**Expected result** 
IMA processing msg should not takes more than 40 seconds in average. getPastEvents should parse logs not longer than 5-7 seconds based on previous results, and results on nodes not affected by this problem.

NOTE: Look closer on logs, attention to PWA thread  on affected chain. 
Compare imaGPE_big.txt with imaIterativeScan13.txt and imaIterativeScan12.txt

[imaGPE_big.txt](https://github.com/skalenetwork/IMA/files/13022989/imaGPE_big.txt)
[imaGPE.txt](https://github.com/skalenetwork/IMA/files/13022986/imaGPE.txt)
[imaIterativeScan12.txt](https://github.com/skalenetwork/IMA/files/13022987/imaIterativeScan12.txt)
[imaIS12.txt](https://github.com/skalenetwork/IMA/files/13022988/imaIS12.txt)
[imaIterativeScan13.txt](https://github.com/skalenetwork/IMA/files/13022983/imaIterativeScan13.txt)
[imaGPE13.txt](https://github.com/skalenetwork/IMA/files/13022984/imaGPE13.txt)

```[tasklist]
### Tasks
- [ ] https://github.com/skalenetwork/IMA/issues/1554
- [ ] https://github.com/skalenetwork/IMA/issues/1554
```

```[tasklist]
### Tasks
- [ ] https://github.com/skalenetwork/IMA/issues/1554
```

### PR DESCRIPTION
**Pull request by sergiy-skalelabs**
IMA event log scanner moved into external module.
No new tests needed.
Legacy network test must be fast.

### CODE REVIEW

File: npms/skale-ima/imaEventLogScan.mjs [hunk 1]
```diff
@@ -86,9 +92,49 @@ export function createProgressiveEventsScanPlan( details, nLatestBlockNumber ) {
...
+        return owaspUtils.toBN( arg.hex );
+    return arg;
+}
+
+export async function safeGetPastEventsProgressiveExternal(
+    details, strLogPrefix, ethersProvider, attempts,
+    joContract, joABI, strEventName,
+    nBlockFrom, nBlockTo, joFilter
+) {
+    if( joABI && typeof joABI == "object" ) {
```
Comment (DmytroNazarenko) [inline npms/skale-ima/imaEventLogScan.mjs:124]: Add log about running external cmd

File: npms/skale-ima/imaSgxExternalSigner.mjs [hunk 2]
```diff
@@ -6,7 +6,7 @@ import * as rpcCall from "../../agent/rpcCall.mjs";
 
 const gIsDebugLogging = false; // development option only, must be always false
 const isColors = owaspUtils.toBoolean( process.argv[2] );
-cc.enable( true );
+cc.enable( false );
```
Comment (DmytroNazarenko) [inline npms/skale-ima/imaSgxExternalSigner.mjs:9]: What is the purpose of this?

**Review by DmytroNazarenko — CHANGES_REQUESTED** [review]

File: npms/skale-ima/imaEventLogScan.mjs (hunk 1 above)
Comment (sergiy-skalelabs) [reply npms/skale-ima/imaEventLogScan.mjs:124]: Added

File: npms/skale-ima/imaSgxExternalSigner.mjs (hunk 2 above)
Comment (sergiy-skalelabs) [reply npms/skale-ima/imaSgxExternalSigner.mjs:9]: This is another external command, where I turned off colorization by default.
Nevertheless it prints only critical errors, it returns error text as is to caller process also.
Just keeping all colors off everywhere.

**Review by DmytroNazarenko — APPROVED** [review]

**Pull request merged**